src/openrct2/ride/Vehicle.cpp

```diff
@@ -14,575 +14,492 @@
+    2066040965,  2147483647,  2074309916,  1859775393,  1518500249,  1073741824,  555809666,   0,           -2066040965,
+    -2147483647, -2074309916, -1859775393, -1518500249, -1073741824, -555809666,  1073741824,  1859775393,  2147483647,
+    1859775393,  1073741824,  -1073741824, -1859775393, -2147483647, -1859775393, -1073741824, 1073741824,  1859775393,
+    2147483647,  1859775393,  1073741824,  -1073741824, -1859775393, -2147483647, -1859775393, -1073741824, 112390610,
+    187165532,   -112390610,  -187165532,  224473165,   -224473165,  303325208,   600568389,   1618265062,  -303325208,
+    -600568389,  -1618265062, -817995863,  -1390684831, -1827693544, 369214930,
 };
 
 /** rct2: 0x009A39C4 */
 static const sint32 Unk9A39C4[] = {
-    2147483647,
-    2096579710,
-    1946281152,
-    2096579710,
-    1946281152,
-    1380375879,
-    555809667,
-    -372906620,
-    -1231746017,
-    -1859775391,
-    1380375879,
-    555809667,
-    -372906620,
-    -1231746017,
-    -1859775391,
-    0,
-    2096579710,
-    1946281152,
-    2096579710,
-    1946281152,
+    2147483647, 2096579710, 1946281152, 2096579710,  1946281152,  1380375879, 555809667,  -372906620, -1231746017, -1859775391,
+    1380375879, 555809667,  -372906620, -1231746017, -1859775391, 0,          2096579710, 1946281152, 2096579710,  1946281152,
 };
 
 static const LocationXY16 AvoidCollisionMoveOffset[] = {
-    { -1,  0 },
-    {  0,  1 },
-    {  1,  0 },
-    {  0, -1 },
+    { -1, 0 },
+    { 0, 1 },
+    { 1, 0 },
+    { 0, -1 },
 };
 
 static bool vehicle_move_info_valid(sint32 cd, sint32 typeAndDirection, sint32 offset)
 {
-    if (cd >= countof(gTrackVehicleInfo)) {
+    if (cd >= static_cast<sint32>(Util::CountOf(gTrackVehicleInfo)))
+    {
         return false;
     }
     sint32 size = 0;
-    switch (cd) {
-    case 0: size = 1024; break;
-    case 1: size = 692; break;
-    case 2: size = 404; break;
-    case 3: size = 404; break;
-    case 4: size = 404; break;
-    case 5: size = 208; break;
-    case 6: size = 208; break;
-    case 7: size = 208; break;
-    case 8: size = 208; break;
-    case 9: size = 824; break;
-    case 10: size = 824; break;
-    case 11: size = 824; break;
-    case 12: size = 824; break;
-    case 13: size = 824; break;
-    case 14: size = 824; break;
-    case 15: size = 868; break;
-    case 16: size = 868; break;
-    }
-    if (typeAndDirection >= size) {
+    switch (cd)
+    {
+    case 0:
+        size = 1024;
+        break;
+    case 1:
+        size = 692;
+        break;
+    case 2:
+        size = 404;
+        break;
+    case 3:
+        size = 404;
+        break;
+    case 4:
+        size = 404;
+        break;
+    case 5:
+        size = 208;
+        break;
+    case 6:
+        size = 208;
+        break;
+    case 7:
+        size = 208;
+        break;
+    case 8:
+        size = 208;
+        break;
+    case 9:
+        size = 824;
```

Comment:
Case label overflow is a very useful property of switch statements.

Reply:
Do you want me to use the older form of this switch? Wouldn't this violate our code style guidelines?

Reply:
No, of course not. I want you to use case label overflow.

Like this:
```C
    switch (cd)
    {
    case 0:
        size = 1024;
        break;
    case 1:
        size = 692;
        break;
    case 2:
    case 3:
    case 4:
        size = 404;
        break;
    case 5:
    case 6:
    case 7:
    case 8:
        size = 208;
        break;
    case 9:
    case 10:
    case 11:
    case 12:
    case 13:
    case 14:
        size = 824;
        break;
    case 15:
    case 16:
        size = 868;
        break;
    }
```

Reply:
Right. That makes perfect sense and I wonder how did I miss that :P